src/__tests__/users.int.test.ts

```diff
@@ -64,12 +65,20 @@ describe('/users', () => {
 			});
 		});
 
+		it('returns 400 when an invalid keycloakUserId is provided', async () => {
+			await request(app)
+				.get(`/users?keycloakUserId=thisisnotauuid`)
+				.set(authHeaderWithAdminRole)
+				.expect(400);
+		});
+
 		it('returns according to pagination parameters', async () => {
 			const { count: baseUserCount } = await loadTableMetrics('users');
-			await Array.from(Array(20)).reduce(async (p, _, i) => {
+			const uuids = Array.from(Array(20)).map(() => uuidv4());
+			await uuids.reduce(async (p, uuid) => {
 				await p;
 				await createUser({
-					authenticationId: `user-${i + 1}`,
+					keycloakUserId: uuid,
 				});
 			}, Promise.resolve());
```

Comment:
`forEach` does not allow you to halt processing to wait to call one iteration until the other has halted; this pattern lets us `await` the previous promise before continuing, which is important since order of insertion determines which created users will be returned.

Reply:
Oh, yes thank you, now I remember the conversations we've had about this before!

Reply:
FWIW I don't like it either :(